omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/ModuleInstallResource1_8.java

```diff
@@ -0,0 +1,154 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.webservices.rest.web.v1_0.resource.openmrs1_8;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.List;
+import javax.servlet.ServletContext;
+import org.springframework.util.ResourceUtils;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.FilenameUtils;
+import org.openmrs.module.Module;
+import org.openmrs.module.ModuleUtil;
+import org.openmrs.module.webservices.helper.ModuleInstall;
+import org.openmrs.module.webservices.helper.ModuleFactoryWrapper;
+import org.openmrs.module.webservices.rest.SimpleObject;
+import org.openmrs.module.webservices.rest.web.ConversionUtil;
+import org.openmrs.module.webservices.rest.web.RequestContext;
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.openmrs.module.webservices.rest.web.annotation.Resource;
+import org.openmrs.module.webservices.rest.web.representation.Representation;
+import org.openmrs.module.webservices.rest.web.resource.api.Creatable;
+import org.openmrs.module.webservices.rest.web.resource.impl.BaseDelegatingResource;
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingResourceDescription;
+import org.openmrs.module.webservices.rest.web.response.IllegalRequestException;
+import org.openmrs.module.webservices.rest.web.response.ResourceDoesNotSupportOperationException;
+import org.openmrs.module.webservices.rest.web.response.ResponseException;
+
+@Resource(name = RestConstants.VERSION_1 + "/moduleinstall", supportedClass = ModuleInstall.class, supportedOpenmrsVersions = {
+        "1.10.*", "1.11.*", "1.12.*", "1.8.*", "1.9.*", "2.0.*", "2.1.*" })
+public class ModuleInstallResource1_8 extends BaseDelegatingResource<ModuleInstall> implements Creatable {
```

Comment:
Instead of creating a new resource, why don't you just use this existing resource by adding an install action? https://github.com/openmrs/openmrs-module-webservices.rest/blob/master/omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/ModuleActionResource1_8.java

Reply:
I wanted to use the existing resource but it did not pass the delegate of the install URL and was passing an array for the modules. I can modify the delegates but wouldn't that affect the existing functionality.

Reply:
I do not seem to understand. Can you explain more and give details?

Reply:
Basing on the delegates passed at this [line](https://github.com/pgutkowski/openmrs-module-webservices.rest/blob/460018808b7bbff61d3405ea916346b1370b84c8/omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/ModuleActionResource1_8.java#L205-L212), modules takes an array and I would have to add a delegate for install URI like what I did [here](https://github.com/malmike/openmrs-module-webservices.rest/blob/859017554af5a14720f5f5b32d47226471c433fa/omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/ModuleInstallResource1_8.java#L135-L141). It would also mean that I would have to modify the module action class. Am not sure how the changes would affect the existing functionality and I found it easier to create a new resource. But I will create a new branch that adds the update functionality to the ModuleActionResource1_8 and see how it pans out.

Reply:
Yes of course you need to modify the switch statement in ModuleActionResource1_8 by including a case for installation.

Reply:
Okay, will be working on the changes